apps/server/src/modules/board/uc/element.uc.ts

```diff
@@ -54,7 +55,9 @@ export class ElementUc extends BaseUc {
 		if (isSubmissionItem(parent)) {
 			await this.checkSubmissionItemWritePermission(userId, parent);
 		} else {
-			await this.checkPermission(userId, element, Action.write);
+			// TODO: fix this temporary hack to prevent students from deleting the DrawingElement
+			const requiredRole = isDrawingElement(element) ? UserRoleEnum.TEACHER : undefined;
```

Comment:
if (isSubmissionItem(parent)) 
if else (isDrawingElement(element))
const requiredRole
await this.checkPermission
else 
await this.checkPermission

Reply:
fixed